src/routes/add-custom-discipline/components/QRCodeReaderOverlay.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react'
+import { AppState, Linking, Modal, PermissionsAndroid } from 'react-native'
+import { BarCodeReadEvent, RNCamera } from 'react-native-camera'
+import styled from 'styled-components/native'
+
+import { CloseCircleIconBlue, CloseCircleIconWhite } from '../../../../assets/images'
+import Button from '../../../components/Button'
+import { BUTTONS_THEME } from '../../../constants/data'
+import labels from '../../../constants/labels.json'
+
+const Container = styled.View`
+  flex: 1;
+  background-color: ${props => props.theme.colors.white};
+`
+
+const Icon = styled.TouchableOpacity`
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  width: 30px;
+  height: 30px;
+`
+
+const Camera = styled(RNCamera)`
+  flex: 1;
+  position: relative;
+  margin: 50px 0 0;
+`
+
+const NotAuthorisedView = styled.View`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+  background-color: ${props => props.theme.colors.lunesWhite};
+`
+
+const NoAuthDescription = styled.Text`
+  font-family: ${props => props.theme.fonts.contentFontRegular};
+  font-size: ${props => props.theme.fonts.defaultFontSize};
+  color: ${props => props.theme.colors.lunesGreyMedium};
+  padding-bottom: 20px;
+`
+
+interface Props {
+  setVisible: (input: boolean) => void
+  setCode: (input: string) => void
+}
+
+const AddCustomDisciplineScreen = ({ setVisible, setCode }: Props): ReactElement => {
+  const appState = useRef(AppState.currentState)
```

Comment:
Optional: To avoid having to use `.current` all the time you could just `useState` (which is interchangeable here) and `useRef` is only syntactic sugar:
```suggestion
  const [appState, setAppState] = useState(AppState.currentState)
```

Reply:
Tested, does not work. I guess because setAppState causes a rerender then.